sre-recipes/recipe_runner.py

```diff
@@ -0,0 +1,196 @@
+            utils.run_interactive_multiple_choice(
+                "Which service has an issue?",
+                affected_service_config["choices"],
+                affected_service_config["answer"])
+
+        incident_cause_config = verify_config.get("incident_cause", {})
+        if incident_cause_config:
+            if "answer" not in incident_cause_config:
+                raise ValueError(
+                    "Correct answer is not specified for incident cause quiz.")
+            elif "choices" not in incident_cause_config:
+                raise ValueError(
+                    "No answer choices configured in incident cause quiz.")
+            utils.run_interactive_multiple_choice(
+                "What was the cause of the issue?",
+                incident_cause_config["choices"],
+                incident_cause_config["answer"])
+
+    ########################## Recipe Action Handlers ##########################
+
+    def __handle_actions(self, actions):
+        """
+        Dispatch and handle a list of actions synchronously.
+
+        Paramters
+        ---------
+        actions: a list of dictionary of paramters.
+            Example: [{'run': 'echo "Hello World!"'}]
+        """
+        loadgen_ip = None
+
+        for action in actions:
+            if "run" in action:
+                output, err = utils.run_shell_command(action["run"])
+                if err:
+                    raise RuntimeError(f"Failed to run action {action}: {err}")
+            elif "loadgen" in action:
+                if not loadgen_ip:
+                    loadgen_ip, err = utils.get_loadgen_ip()
+                    if err:
+                        raise RuntimeError(f"Failed to get loadgen IP: {err}")
+                if action["loadgen"] == "stop":
+                    resp = requests.post(f"http://{loadgen_ip}:81/api/stop")
+                    if not resp.ok:
+                        raise RuntimeError(
+                            f"Failed to stop existing load generation: {resp.status_code} {resp.reason}")
+                elif action["loadgen"] == "spawn":
+                    user_type = action.get(
+                        "user_type", DEFAULT_LOADGEN_USER_TYPE)
+                    resp = requests.post(
+                        f"http://{loadgen_ip}:81/api/spawn/{user_type}",
+                        {
+                            "user_count": int(action.get("user_count", DEFAULT_LOADGEN_USER_COUNT)),
+                            "spawn_rate": int(action.get("spawn_rate", DEFAULT_LOADGEN_SPAWN_RATE)),
+                            "stop_after": int(action.get("stop_after", DEFAULT_LOADGEN_TIMEOUT_SECONDS))
+                        })
+                    if not resp.ok:
+                        raise RuntimeError(
+                            f"Failed to start load generation: {resp.status_code} {resp.reason}")
+            else:
+                raise NotImplementedError(f"action not supported: {action}")
```

Comment:
This function is a bit long, and will be hard to maintain if we add more actions. I'd recommend breaking it up into different helper functions.

If you take my suggestion above and add a `type` key, you can easily load the helper with a dictionary:

```
action_map = {'run', _runfunc, 'loadgen': _loadgen_func}

for action in actions:
   helper = action_map[action['type']
   helper(action)
...
  
```

Reply:
Done. Used a new `ActionHandler` class to encapsulate all the logic. Updated README as well